### PR TITLE
nixos/netdata: Disable PrivateTmp setting for Netdata service

### PR DESCRIPTION
**Pull request by leotaku**
###### Motivation for this change

This should fix the second issue listed in  #127375, namely that the `netdata-claim.sh` script is unable to find and open the Netdata socket.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/monitoring/netdata.nix [hunk 1]
```diff
@@ -203,7 +203,7 @@ in {
         # Sandboxing
         ProtectSystem = "full";
         ProtectHome = "read-only";
-        PrivateTmp = true;
+        PrivateTmp = false;
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/netdata.nix:206]: Please add a comment why this needs to be false.

**Comment by aanderse** [comment]
Is the socket supposed to go under `/run` instead?

**Comment by leotaku** [comment]
@aanderse The socket path is currently hard-coded in the upstream Netdata project. So while it's might not be supposed to be in `/tmp` from a FHS standpoint, it's definitely supposed to be there when talking about Netdata.

https://github.com/netdata/netdata/blob/d2030d0241ec30a372c84656ff6a7bc3fe0c9e81/daemon/commands.h#L9

**Comment by aanderse** [comment]
@leotaku thanks for pointing that out.

**Comment by wmertens** [comment]
Would it not be better to patch it so it uses /run?

**Comment by leotaku** [comment]
> Would it not be better to patch it so it uses /run?

I disagree. Other software that is not distributed with Netdata might want to communicate with it using the IPC socket. I would also generally like to avoid distro-specific patches when they are not absolutely necessary.

**Comment by wmertens** [comment]
Hmm - mysql uses `/run/mysql/mysql.sock` while most documentation thinks it's under `/var/lib`.

OTOH that's a configuration option. I will open an issue on the netdata repo

**Comment by Mic92** [comment]
@Izorkin please have a look.

**Comment by Mic92** [comment]
> @aanderse The socket path is currently hard-coded in the upstream Netdata project. So while it's might not be supposed to be in `/tmp` from a FHS standpoint, it's definitely supposed to be there when talking about Netdata.
> 
> https://github.com/netdata/netdata/blob/d2030d0241ec30a372c84656ff6a7bc3fe0c9e81/daemon/commands.h#L9

This is actually dangerous. `/tmp` is world-writable. An attacker could put a symlink in this location and override an unrelated file.

**Comment by wmertens** [comment]
@Mic92 good point - I already requested it at https://community.netdata.cloud/t/make-agent-cli-pipe-configurable/1448 but security vulnerabilities are a separate flow.

**Comment by Mic92** [comment]
> @Mic92 good point - I already requested it at https://community.netdata.cloud/t/make-agent-cli-pipe-configurable/1448 but security vulnerabilities are a separate flow.

Email is sent. If I don't come back with an update within 1 weeks time, please ping me again.

**Comment by leotaku** [comment]
Ping @Mic92

**Comment by Mic92** [comment]
They acknowledged the bug but they have not provided a change yet. This patch should work for us: https://github.com/Mic92/netdata/commit/c9fc2b7bf5fdfc5d1ca01284aad86ea97c21bccc
Since our systemd configuration will make sure `/run/netdata` exists and is writeable. I don't think I can upstream this change as it is since it might not work on other platforms netdata supports but I think it is a good start.

**Comment by wmertens** [comment]
Right, ideally they'd make the pipe path part of the build config

**Comment by leotaku** [comment]
Right, how would you like me to proceed? Should I force-push your suggested changes, revert mine and then apply yours, or create an entirely new PR?

**Comment by Mic92** [comment]
> Right, how would you like me to proceed? Should I force-push your suggested changes, revert mine and then apply yours, or create an entirely new PR?

Just force-push the patch here.

**Comment by leotaku** [comment]
@Mic92 Done

File: pkgs/tools/system/netdata/default.nix [hunk 2]
```diff
@@ -42,6 +42,10 @@ in stdenv.mkDerivation rec {
     # required to prevent plugins from relying on /etc
     # and /var
     ./no-files-in-etc-and-var.patch
+    # required to put IPC socket in globally accessible
+    # directory, until a proper solurion is provided
```
Comment (Mic92) [inline pkgs/tools/system/netdata/default.nix:46]:
```suggestion
    # directory, until a proper solution is provided
```

File: pkgs/tools/system/netdata/default.nix [hunk 3]
```diff
@@ -42,6 +42,10 @@ in stdenv.mkDerivation rec {
...
+    # upstream
```
Comment (Mic92) [inline pkgs/tools/system/netdata/default.nix:47]:
```suggestion
    # The current IPC location is unsafe as it writes
    # a fixed path in /tmp, which is world-writable.
    # Therefore we put it into `/run/netdata`, which is owned
    # by netdata only.
```

**Pull request merged**